src/python/WMCore/ReqMgr/DataStructs/Request.py

```diff
@@ -196,15 +196,18 @@ def _maskTaskStepChain(self, prop, chain_name, default=None):
                 chain_key = "%s%s" % (chain_name, i + 1)
                 chain = self.data[chain_key]
                 if prop in chain:
-                    result.add(chain[prop])
+                    foundValue = chain[prop]
                 else:
                     if isinstance(defaultValue, dict):
-                        value = defaultValue.get(chain_key, None)
+                        foundValue = defaultValue.get(chain_key, None)
                     else:
-                        value = defaultValue
+                        foundValue = deepcopy(defaultValue)
 
-                    if value is not None:
-                        result.add(value)
+                if foundValue not in [None, ""]:
```

Comment:
Such string with a whitespace char in it would not be a default for anything in the system. So, if `" "` is found, we have to consider that as a legit user input.

Reply:
Thanks @amaltaro . That's good, but still are we expecting such value to ever come into the query and do we actually care about those use cases at all. I am asking because if we don't ... then no need for stopping this PR any more. We just make a comment and continue.

Reply:
Workflow spec should have many validation functions, enforcing a spec to follow a given schema and data type.
However, it could be that something sneaks in. 

For instance (not sure it would ever be allowed, but ...), if one set `PrepID=" "` in a request dictionary, this function must add that value to the results returned to the user, given that that is a legitimate input. In other words, we do not want to filter out a string that evaluates to true: `" "`, different than an empty string which will evaluate to false: `""`.

Reply:
I see. Then we want to treat nonempty whitespaced strings as `True` while zero length ones as `False`. So in this case I would say we are in total consensus with the Python convention for casting to `bool`, then why not just:

```
if foundValue:
    ...
```

Reply:
Except, maybe if we want to abstract from the cases where the `faoundValue` itself is `True` or `False`.

Reply:
> Except, maybe if we want to abstract from the cases where the foundValue itself is True or False.

Exactly. False must be considered, hence the somehow weird `not in` statement that I put in place.

Reply:
Super, all clear then. Thanks @amaltaro.